### ISSUE TITLE
Addon Blist Error

### ISSUE THREAD

**Issue by lewyo** [Windower/Issues#931]
Today was the first time i tried this addon and when i try to load it i get his error message : 
Blist: Lua runtime error: ...rogram Files (x86)/Windower4//addons/libs/logger.lua:77 attempt to index field 'settings' (a nil value)

Any help on a fix for this addon would be greatly appreciated.

**Comment by z16** [comment]
I updated the file in a number of ways, but since I never used it, I'm not sure how well it works. So before I push it live, can you do me a favor and get it from [here](https://gist.github.com/z16/9eecc7083bf03ba6424ddada2274c9c7/raw/b23ddac045d7276b3bb5acb0f5176be24e8ab792/blist.lua) and let me know how it works? You can right-click the link and select "Save file" or "Save link" or whatever your browser calls it. Then save it to `Windower/addons/blist/`, it should replace the `blist.lua` file in there. Then try reloading it again by typing `//lua load blist` and let me know if it works.

**Comment by lewyo** [comment]
Thanks for your quick reply, But sadly when i try to //lua load blist it is still giving me the same error  after replacing the file.

**Comment by joshk6656** [comment]
What OS are you on? Have you done anything to prevent windower from
updating? I am thinking your libs may not be up to date. But I'll let arcon
decide.

On Jun 1, 2017 3:23 PM, "lewyo" <notifications@github.com> wrote:

Thanks for your quick reply, But sadly when i try to //lua load blist it is
still giving me the same error after replacing the file.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Windower/Issues/issues/931#issuecomment-305609679>, or mute
the thread

**Comment by lewyo** [comment]
I am on Win 7 64 and i have never blocked win form updating.

**Comment by z16** [comment]
Can you check the `Windower/addons/libs/` folder for a file called `logger.xml`. Is it there? If so, what are its contents?

**Comment by lewyo** [comment]
<?xml version="1.1" ?>
<settings>
    <global>
        <logtofile>false</logtofile>        
        <defaultfile>lua.log</defaultfile>    
        <logcolor>207</logcolor>            
        <errorcolor>167</errorcolor>        
        <warningcolor>200</warningcolor>    
        <noticecolor>160</noticecolor>        
    </global>
</settings>

**Comment by lewyo** [comment]
Is the logger XML file in my last post correct on what it should look like?

**Comment by z16** [comment]
Yes, it looks fine. I'm still debating what could be causing it. Unfortunately I can't run FFXI from here, but I'll be back in about a week, then I can test this myself.

**Comment by lewyo** [comment]
OK thank you.

**Comment by lewyo** [comment]
Z16 have you had a chance to test this out yet?

**Comment by z16** [comment]
@lewyo Sorry, I didn't get the notice, and due to time constraints lately I haven't been checking the tracker as carefully as I used to. To mention me properly prepend an `@` symbol to my name.

I did not look into it yet, sorry. But I can do it now. First I'd like to ask if you experience similar issues with other addons, or just *blist*? For example, can you add the *Treasury* addon, if you're not already using it, and try to load it, then type: `tr lot list`

This won't have any actual effect but it should output something to your chatlog. If this works, the error is isolated to *blist*.

**Comment by joshk6656** [comment]
I tested this command on characters with and without a lot list and it does
properly print the list to the chat log

**Comment by z16** [comment]
I just noticed an unrelated error in the addon that means it can't work right... it's trying to include an old file that doesn't exist anymore. Although it does not explain your described error, @lewyo try [this file](https://gist.githubusercontent.com/z16/5f12da47304c3243351cc2e3faf22457/raw/b07e7aefa79479152d8ace9f457210d0eb5bbea0/gistfile1.txt) anyway and see if that helps.

**Comment by Sechs** [comment]
Tested last version and found the following errors:

1) if the syntax command isn't perfectly exact, it will give an error on line 99 (it's the line supposed to generate a message saying to "correct" the syntax)
2) I've performed several tests, but even if the name appears in the list, the addon simply doesn't work (read below for test details)
3) If I add a name to the list and reload the addon, that name will be in lower case, not sure if this affects the addon or not (is it case sensitive?)
4) Adding the same name with different case, is considered a different entry in the list


These are the things I tested:
1) Day 1, day 0, day 2
2) normal case name (Lewyo), downcased name (lewyo)
3) Soft Mode, Hard Mode, Muted mode
4) Tried in party chat, Linkshell chat, direct tells

I combined all of these so I could test all of the above in all possible combinations.
All messages were getting through regardless of the addon being properly configured with the name viewable in the list.

**Comment by Sechs** [comment]
@z16 please keep us updated if there are news of any sort to this addon, thanks! ^^
